pkg/server/endpoints/middleware.go

```diff
@@ -111,16 +127,16 @@ func AgentAuthorizer(log logrus.FieldLogger, ds datastore.DataStore) middleware.
 			return status.Errorf(codes.Internal, "unable to look up agent information")
 		case resp.Node == nil:
 			log.Error("Agent is not attested")
-			return status.Errorf(codes.PermissionDenied, "agent %q is not attested", id)
+			return permissionDenied(types.PermissionDeniedDetails_AGENT_NOT_ATTESTED, "agent %q is not attested", id)
```

Comment:
it may not be immediately obvious to the receiver what "not attested" means... maybe `no attestation entry exists for agent %q` ?

Reply:
hmm. there seems to be a tension on all of these messages about conveying the high level concept "agent is not attested" (or in other words, is not a valid agent in the trust domain) and how the server determines that it isn't attested ("no entry exists"). I can see using the later for ease of debugging but worry a little about leaking server implementation details back to the client.

Reply:
yeah I guess I was thinking from the perspective of the debugger, but it seems this change is more focused on enabling the agent to make better decisions based on the result of the server, so that the operator doesn't need to do anything anyway.. i'm cool with the existing language